zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateBusinessRuleTaskTest.java

```diff
@@ -0,0 +1,187 @@
+import io.camunda.zeebe.test.util.BrokerClassRuleHelper;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestWatcher;
+
+public class MigrateBusinessRuleTaskTest {
+
+  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+  private static final String DMN_RESOURCE = "/dmn/decision-table.dmn";
+  @Rule public final TestWatcher watcher = new RecordingExporterTestWatcher();
+  @Rule public final BrokerClassRuleHelper helper = new BrokerClassRuleHelper();
+
+  @Test
+  public void shouldMigrateCalledDecisionBasedBusinessRuleTaskWithAnIncident() {
+    // given
+    final String sourceProcessId = helper.getBpmnProcessId();
+    final String targetProcessId = helper.getBpmnProcessId() + "2";
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlClasspathResource(DMN_RESOURCE)
+            .withXmlResource(
+                Bpmn.createExecutableProcess(sourceProcessId)
+                    .startEvent()
+                    .businessRuleTask(
+                        "businessRuleTask1",
+                        t ->
+                            t.zeebeCalledDecisionId("nonExistingDecision")
+                                .zeebeResultVariable("result"))
+                    .endEvent()
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .startEvent()
+                    .businessRuleTask(
+                        "businessRuleTask2",
+                        t ->
+                            t.zeebeCalledDecisionId("jedi_or_sith")
+                                .zeebeResultVariable("jedi_or_sith"))
+                    .endEvent("target_process_end")
+                    .done())
+            .deploy();
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(sourceProcessId)
+            .withVariable("lightsaberColor", "green")
+            .create();
+
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetProcessId);
+
+    RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .withElementId("businessRuleTask1")
+        .await();
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("businessRuleTask1", "businessRuleTask2")
+        .migrate();
+
+    RecordingExporter.processInstanceMigrationRecords(ProcessInstanceMigrationIntent.MIGRATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .await();
```

Comment:
🔧 This is redundant. The call to `.migrate()` (the last line above this) already awaits this exact event. 
🔧 We can of course be more explicit by awaiting the Incident to be migrated, see: d90e8d4986254bbb542ba5415aa153c6b1bdc8d4